src/directives/class-map.ts

```diff
@@ -66,55 +48,44 @@ export const classMap = directive((classInfo: ClassInfo) => (part: Part) => {
 
   let previousClasses = previousClassesCache.get(part);
   if (previousClasses === undefined) {
-    previousClasses = new Map();
-    previousClassesCache.set(part, previousClasses);
-    // Normalize all static classes into individual tokens. This is necessary
-    // since each individual string could contain multiple tokens.
-    const strings = committer.strings.join(' ');
-    let match;
-    while ((match = tokens.exec(strings))) {
-      // Ensure static classes are never removed, by setting them to true
-      previousClasses!.set(match[0], true);
-    }
-    changed = true;
+    // Write static classes once
+    // Use setAttribute() because className isn't a string on SVG elements
+    element.setAttribute('class', committer.strings.join(' '));
+    previousClassesCache.set(part, previousClasses = new Set());
   }
 
+  // IE11 doesn't support classList on SVG elements, so we emulate it with a Set
+  const classes = new Set((element.getAttribute('class') || '').split(/\s+/));
+
   // Remove old classes that no longer apply
   // We use forEach() instead of for-of so that re don't require down-level
   // iteration.
-  previousClasses.forEach((value: unknown, name: string) => {
-    // If the value is true, then it was a static class, which we do not remove
-    // unless the ClassInfo specifically overrides it.
-    if (value !== true && !(name in classInfo)) {
-      changed = true;
+  previousClasses.forEach((name) => {
+    if (!(name in classInfo)) {
+      classes.delete(name);
       previousClasses!.delete(name);
+      changed = true;
     }
   });
 
-  // Add or remove classes based on their ClassInfo value
+  // Add or remove classes based on their classMap value
   for (const name in classInfo) {
     const value = classInfo[name];
-    const cached = previousClasses.get(name);
     // We explicitly want a loose truthy check of `value` because it seems more
```

Comment:
This comment now applies to the `if (value)` statement inside this if statement.